vnc/vnc.c

```diff
@@ -53,11 +53,29 @@ void DEFAULT_CC
 rfbEncryptBytes(char *bytes, char *passwd)
 {
     char key[24];
+    void *des;
+    int len;
+
+    /* key is simply password padded with nulls */
+    g_memset(key, 0, sizeof(key));
+    len = MIN(g_strlen(passwd), 8);
+    g_mirror_memcpy(key, passwd, len);
+    des = ssl_des3_encrypt_info_create(key, 0);
+    ssl_des3_encrypt(des, 8, bytes, bytes);
+    ssl_des3_info_delete(des);
+    des = ssl_des3_encrypt_info_create(key, 0);
+    ssl_des3_encrypt(des, 8, bytes + 8, bytes + 8);
+    ssl_des3_info_delete(des);
+}
+
+/******************************************************************************/
+/* taken from vncauth.c */
```

Comment:
Function comments should primarily describe what the function does, not where the code came from.